.github/workflows/ci.yaml

```diff
@@ -129,3 +129,20 @@ jobs:
 
       - name: Clippy check
         run: cargo clippy --features=linux-native -- -D warnings
+
+  doc_build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Fetch head
+        uses: actions/checkout@v4
+
+      - name: Install rust nightly
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: nightly
+          targets: x86_64-unknown-linux-gnu, aarch64-apple-darwin, aarch64-apple-ios, x86_64-pc-windows-msvc
+          components: rust-src
```

Comment:
Didn't realize `Cargo.lock` is not there, this should fix the failed CI.

Reply:
Yes, because this is a library, there's no Cargo.lock.

What do you think about moving the docs build out of the primary CI script and restricting it to just run on PRs and via explicit request, rather than on all pushes? There are really very few changes that can break the docs build, and it's an unnecessary overhead on most push requests.

Reply:
Yes, I think that'll be better or even not having CI (can always run test script manually before release). Maybe move it to `publish.yml` to make sure the docs.rs will success? As you said, very few changes will break the docs build, so I am open to all options.

Reply:
Please also check the new commit I added https://github.com/hwchen/keyring-rs/pull/237/commits/8a5c6dd838c701fd72cdf17cf4d10328d412e461
It failed in linux target because lack of libdbus dependency, but the failure not catch in bash run. I add `set -e` to make it fast failed.

Reply:
I don't like the idea of having it in publish, because that's only run at release time. I'd rather have it be separate and run on pull requests, because I always pull to master before I release. Can you please move it to a separate workflow (called `doctest`)? Also I think if you make that a matrix workflow for all the desired platforms then you may not need a shell script? I think you might be able to just have the command line for the matrix build be what gets run in the loop in the current shell script.

Reply:
> I'd rather have it be separate and run on pull requests, because I always pull to master before I release. Can you please move it to a separate workflow (called doctest)?

Sounds good, I like the idea :) 
Changed. 

> Also I think if you make that a matrix workflow for all the desired platforms then you may not need a shell script?

Using matrix might be bring overhead because for each target in the matrix it is an independent run. I move the bash script content into CI directly, which is more secure I think. Since otherwise changing script may trigger unexpected CI run.